stackstorm-compose/docker-compose.yml

```
@@ -0,0 +1,219 @@
+    image: ${ST2_IMAGE_REPO:-stackstorm/}st2actionrunner:${ST2_VERSION:-3.3dev}
+    restart: on-failure
+    depends_on: ["st2api"]
+    networks:
+      - private
+    volumes:
+      - ./st2.docker.conf:/etc/st2/st2.docker.conf:ro
+      - ./st2.user.conf:/etc/st2/st2.user.conf:ro
+      - stackstorm-packs-configs:/opt/stackstorm/configs:rw
+      - stackstorm-packs:/opt/stackstorm/packs:rw
+      - ${ST2_PACKS_DEV:-./packs.dev}:/opt/stackstorm/packs.dev:rw
+      - stackstorm-virtualenvs:/opt/stackstorm/virtualenvs:rw
+      - stackstorm-ssh:/home/stanley.ssh
+    dns_search: .
+  st2garbagecollector:
+    image: ${ST2_IMAGE_REPO:-stackstorm/}st2garbagecollector:${ST2_VERSION:-3.3dev}
+    restart: on-failure
+    depends_on: ["st2api"]
+    networks:
+      - private
+    volumes:
+      - ./st2.docker.conf:/etc/st2/st2.docker.conf:ro
+      - ./st2.user.conf:/etc/st2/st2.user.conf:ro
+    dns_search: .
+  st2notifier:
+    image: ${ST2_IMAGE_REPO:-stackstorm/}st2notifier:${ST2_VERSION:-3.3dev}
+    restart: on-failure
+    depends_on: ["st2api"]
+    networks:
+      - private
+    volumes:
+      - ./st2.docker.conf:/etc/st2/st2.docker.conf:ro
+      - ./st2.user.conf:/etc/st2/st2.user.conf:ro
+    dns_search: .
+  st2resultstracker:
+    image: ${ST2_IMAGE_REPO:-stackstorm/}st2resultstracker:${ST2_VERSION:-3.3dev}
+    restart: on-failure
+    depends_on: ["st2api"]
+    networks:
+      - private
+    volumes:
+      - ./st2.docker.conf:/etc/st2/st2.docker.conf:ro
+      - ./st2.user.conf:/etc/st2/st2.user.conf:ro
+    dns_search: .
+  st2rulesengine:
+    image: ${ST2_IMAGE_REPO:-stackstorm/}st2rulesengine:${ST2_VERSION:-3.3dev}
+    restart: on-failure
+    depends_on: ["st2api"]
+    networks:
+      - private
+    volumes:
+      - ./st2.docker.conf:/etc/st2/st2.docker.conf:ro
+      - ./st2.user.conf:/etc/st2/st2.user.conf:ro
+    dns_search: .
+  st2sensorcontainer:
+    image: ${ST2_IMAGE_REPO:-stackstorm/}st2sensorcontainer:${ST2_VERSION:-3.3dev}
+    restart: on-failure
+    depends_on: ["st2api"]
+    networks:
+      - private
+    dns_search: .
+    volumes:
+      - ./st2.docker.conf:/etc/st2/st2.docker.conf:ro
+      - ./st2.user.conf:/etc/st2/st2.user.conf:ro
+      - stackstorm-virtualenvs:/opt/stackstorm/virtualenvs:ro
+      - stackstorm-packs:/opt/stackstorm/packs:ro
+      - stackstorm-packs-configs:/opt/stackstorm/configs:ro
+      - ${ST2_PACKS_DEV:-./packs.dev}:/opt/stackstorm/packs.dev:ro
+  st2timersengine:
+    image: ${ST2_IMAGE_REPO:-stackstorm/}st2timersengine:${ST2_VERSION:-3.3dev}
+    restart: on-failure
+    depends_on: ["st2api"]
+    networks:
+      - private
+    dns_search: .
+    volumes:
+      - ./st2.docker.conf:/etc/st2/st2.docker.conf:ro
+  st2client:
+    image: ${ST2_IMAGE_REPO:-stackstorm/}st2actionrunner:${ST2_VERSION:-3.3dev}
+    restart: on-failure
+    command: sleep infinity
+    networks:
+      - private
+    environment:
+      ST2_AUTH_URL: ${ST2_AUTH_URL:-http://st2auth:9100/}
```

```suggestion
    environment:
      ST2CLIENT: 1
      ST2_AUTH_URL: ${ST2_AUTH_URL:-http://st2auth:9100/}
```

so we print some useful info when switching into st2client like:
https://github.com/StackStorm/st2-dockerfiles/blob/2f2b85497ed48491775654c4377502d793d8cbd5/base/files/.welcome.sh#L11-L18

---

I like this. It also can be utilized by k8s as well just by setting the environment variable in helm.

---

Yes, this is already in place for K8s/Helm.

---

I added the environment variable, but not sure if we want to change the Dockerfile as part of this growing PR.